gateway/src/apicast/policy/token_introspection/token_introspection.lua

```diff
@@ -26,12 +28,28 @@ function _M.new(config)
       ssl = { verify = resty_env.enabled('OPENSSL_VERIFY') }
     }
   }
+
+  -- If there's no 'max_ttl_tokens', the ttl will be the one returned by the
+  -- token introspection endpoint. To disable caching, max_ttl_tokens needs to
+  -- be set to 0.
+  self.caching_enabled = not config.max_ttl_tokens or config.max_ttl_tokens > 0
+
+  if self.caching_enabled then
+    self.tokens_cache = tokens_cache.new(
+      config.max_ttl_tokens, config.max_cached_tokens)
+  end
+
   return self
 end
 
 --- OAuth 2.0 Token Introspection defined in RFC7662.
 -- https://tools.ietf.org/html/rfc7662
 local function introspect_token(self, token)
+  if self.caching_enabled then
```

Comment:
What about using some noop cache instead? I wonder how this affects LuaJIT.

If this file would be cached then there would be just one instance of this function shared with all policies. When some would have cache and others not then LuaJIT could not compile this.

But because each policy instance is required again every policy has own JITable function. 

So having a "no cache" implementation would be just for readability and not for performance. Polymorphism over conditionals. LuaJIT will then inline those functions to nothing if they are empty.

Reply:
OK. I didn't think about LuaJIT while doing this.

Reply:
Changed this to get rid of the conditionals. Although in this particular case I don't see a big difference.